### PR TITLE
[PLAY-497] Paginate/passthrough

### PR DESCRIPTION
**Pull request by jasoncypret**
Re-work of #2249 

PR Includes: 

1. A kit wrapper for the render
2. Removed all client dependencies to make pagination optional
3. Provided instructions for use
4. [Confirmed it without will_paginate included if not used by client](https://share.getcloudapp.com/YEuDelkD)

TODO List:

1. Confirm with Nitro Rails dev on this solution
2. See if we can get `call` or `eval` working with strategy to clean up code
3. Test out all global props
4. Re-work styles to utilize kit instead of global props
5. Create an alpha
6. Fix the found a hover bug on borders for previous and next buttons

### CODE REVIEW

File: playbook-website/app/controllers/pages_controller.rb [hunk 1]
```diff
@@ -6,6 +6,15 @@
 
 require_relative "application_controller"
 
+# Need Wade's input
+require "will_paginate"
+require "playbook/pagination_renderer"
+# Since we don't require it for clients should we make steps for them...
+# To require it if they want it? Boostrap renderer works this way
+# Nitro Example: https://github.com/powerhome/nitro-web/blob/master/components/nitro_theme/lib/nitro_theme.rb#L8-L10
```
Comment (wadewinningham) [inline playbook-website/app/controllers/pages_controller.rb:14]: My input on what exactly?

File: playbook/app/pb_kits/playbook/pb_paginate/paginate.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: true
+
+module Playbook
+  module PbPaginate
+    class Paginate < ::Playbook::KitBase
+      prop :stragety # Need to add protection
```
Comment (jasperfurniss) [inline playbook/app/pb_kits/playbook/pb_paginate/paginate.rb:6]: Let's correct the spelling of strategy here. 
```suggestion
      prop :strategy # Need to add protection
```

File: playbook/app/pb_kits/playbook/pb_paginate/docs/_paginate_default.html.erb [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+<%= pb_rails("title", props: { size: 4, text: "Playbook Renderer"}) %>
+<%= will_paginate @users, renderer: Playbook::Pagination::Rails %>
+<%= pb_rails("body") do %>
+  You need to also add: <code>require "playbook/pagination_renderer"</code> in your app to use.
+<% end %>
+<br><br>
+
+<%= pb_rails("title", props: { size: 4, text: "Playbook Kit (Passthrough)"}) %>
+<%= pb_rails("paginate", props: { stragety: "will_paginate", model: @users }) %>
```
Comment (jasperfurniss) [inline playbook/app/pb_kits/playbook/pb_paginate/docs/_paginate_default.html.erb:16]:
```suggestion
<%= pb_rails("paginate", props: { strategy: "will_paginate", model: @users }) %>
```

File: playbook/app/pb_kits/playbook/pb_paginate/paginate.html.erb [hunk 4]
```diff
@@ -0,0 +1,5 @@
+<%= will_paginate object.model, renderer: Playbook::Pagination::Rails %>
+
+<%# Couldn't get this working %>
+<%# Need a Rails Dev to look %>
+<%# <%= eval(object.stragety) object.model, renderer: Playbook::Pagination::Rails %>
```
Comment (jasperfurniss) [inline playbook/app/pb_kits/playbook/pb_paginate/paginate.html.erb:5]:
```suggestion
<%# <%= eval(object.strategy) object.model, renderer: Playbook::Pagination::Rails %>
```

File: playbook-website/app/controllers/pages_controller.rb (hunk 1 above)
Comment (jasoncypret) [reply playbook-website/app/controllers/pages_controller.rb:14]: Is this ok to leave to the client to include If they want it? I assume yes but wanted to check @wadewinningham

File: playbook/app/pb_kits/playbook/pb_paginate/paginate.html.erb [hunk 5]
```diff
@@ -0,0 +1,5 @@
+<%= will_paginate object.model, renderer: Playbook::Pagination::Rails %>
+
+<%# Couldn't get this working %>
+<%# Need a Rails Dev to look %>
+<%# <%= eval(object.strategy) object.model, renderer: Playbook::Pagination::Rails %>
```
[the remark this answers is not in the document]
Comment (wadewinningham) [reply playbook/app/pb_kits/playbook/pb_paginate/paginate.html.erb:5]: Re: pagination, I think that's fine but others may have opinions. To me, if someone may not be using the pagination part, then why include the dependency? They can do it themselves if they need it.

[the remark this answers is not in the document]
Comment (wadewinningham) [reply playbook/app/pb_kits/playbook/pb_paginate/paginate.html.erb:5]: I fully support the strategy method you mention. That would be awesome. Carlos may could assist there.

File: playbook/lib/playbook.rb [hunk 6]
```diff
@@ -12,6 +12,8 @@
 require "playbook/kit_base"
 require "playbook/kit_resolver"
 require "playbook/markdown"
+# require "playbook/pagination_renderer"
+# REMOVED TO MAKE THIS AN OPTIONAL INCLUDE FOR CLIENTS
```
Comment (jasperfurniss) [inline playbook/lib/playbook.rb:16]:
```suggestion
```

**Review by jasperfurniss — APPROVED** [review]

**Pull request merged**